### PR TITLE
[docs] add documentation around patch release

### PR DESCRIPTION
**Pull request by codeboten**
This specifically calls out the correct way to merge the pull request.

cc @bogdandrutu as the other recent patch releaser

### CODE REVIEW

File: docs/release.md [hunk 1]
```diff
@@ -125,6 +125,18 @@ When considering making a bugfix release on the `v0.N.x` release cycle, the bug
...
 
+### Bugfix release procedure
+
+The following documents the procedure to release a bugfix
+
+1. Create a pull request against the `release/<release-series>` (e.g. `release/v0.45.x`) branch to apply the fix.
+2. Create a pull request to update version number against the `release/<release-series>` branch.
+3. Once those changes have been merged, create a pull request to the `main` branch from the `release/<release-series>` branch.
+4. Enable the **Merge pull request** setting in the repository's **Settings** tab.
+5. **IMPORTANT**: The pull request to bring the changes from the release branch *MUST* be merged using the **Merge pull request** method, and *NOT* squashed using “**Squash and merge**”. This is important as it allows us to ensure the commit SHA from the release branch is also on the main branch. **Not following this step will cause much go dependency sadness.**
```
Comment (bogdandrutu) [inline docs/release.md:136]: One small caveat that happens, after you "merge pull request" the branch will get deleted :) Just add a note to "restore" the branch.

File: docs/release.md [hunk 2]
```diff
@@ -125,6 +125,18 @@ When considering making a bugfix release on the `v0.N.x` release cycle, the bug
...
+6. Tag all the modules with the new release version by running the `make add-tag` command (e.g. `make add-tag TAG=v0.55.0`). Push them to `open-telemetry/opentelemetry-collector` with `make push-tag TAG=v0.55.0`. Wait for the new tag build to pass successfully.
```
Comment (bogdandrutu) [inline docs/release.md:137]: I would move this as step 4.

**Review by bogdandrutu — APPROVED** [review]

File: docs/release.md (hunk 2 above)
Comment (codeboten) [reply docs/release.md:137]: I thought pushing the tag before it was available on the main branch caused problems. I just tested it (with a test library) and it doesn't, but I seem to recall there were issues w/ pushing tags from a release branch in the past.

Comment (bogdandrutu) [reply docs/release.md:137]: The problem was that we did not use "merge" there, that's what I remember.

**Pull request merged**